nanobenchmarks/function_registry.cpp

```diff
@@ -0,0 +1,42 @@
+#include "nanobenchmarks/function_registry.hpp"
+
+#include <functional>
+#include <map>
+#include <string>
+
+#include "glog/logging.h"
+
+namespace principia {
+namespace nanobenchmarks {
+namespace _function_registry {
+namespace internal {
+
+bool FunctionRegistry::Register(std::string_view name,
+                                BenchmarkedFunction function) {
+  CHECK(singleton_.names_by_function_.emplace(function, name).second)
+      << " Registering function " << function << " as " << name << ": "
+      << "function already registered as "
+      << singleton_.names_by_function_[function];
+  CHECK(singleton_.functions_by_name_.emplace(name, function).second)
+      << " Registering function " << function << " as " << name << ": "
+      << " name already taken by "
+      << singleton_.functions_by_name_.find(name)->second;
+  return true;
+}
+
+FunctionRegistry& FunctionRegistry::singleton_ = *new FunctionRegistry();
+
+std::map<std::string, BenchmarkedFunction, std::less<>> const&
```

Comment:
Use absl maps everywhere.

Reply:
These are ordered, so that the tests show up in alphabetical(ish) order; we don’t use absl ordered maps unless we have a good reason to.